code/software/libs/src/start_serial/init.S

```diff
@@ -60,9 +63,57 @@ POSTINIT:                         ; running from copied run addr location
     dbra    D1,.COPY_LOOP         ; outer loop
 
 PREMAIN:
-    move.l  $0004,-(A7)           ; push (soft) reset vector if kmain returns
     lea.l   __kinit,A0
     jsr     (A0)                  ; prepare C environment
+    bsr.s   CALL_CTORS            ; Call global constructors
     lea.l   kmain,A0
-    jmp     (A0)                  ; Fly user program, Fly!
+    jsr     (A0)                  ; Fly user program, Fly!
+    move.l  #0,-(A7)              ; Call __cxa_atexit with NULL dso_handle

```

Comment:
A very minor thing, but I think this comment should read `__cxa_finalize`.

Reply:
> Very cool work! It's nice to see it working! One little comment I have is that currently `exit()` just warm-boots, while _I think_ it should call the destructors. Is that something you think should be added, or are good without for now?

Ah yes, excellent point, good spot. I'll look into a way to make this happen cleanly 👍

Reply:
> A very minor thing, but I think this comment should read `__cxa_finalize`.

Yup, it should. Will fix, thanks :)

Reply:
Having looked into `exit` calling destructors, I've found that it's a bit tricky. For that reason, I'm proposing that we wait until `r2-initial-import` is merged before fixing (and accept that it will not work as planned with this merge - to be fair, it's already broken with HUGEROM, as noted at the bottom of the ticket linked below).

I've raised https://github.com/rosco-m68k/rosco_m68k/issues/269 to cover fixing this properly (post this merge and `r2-initial-import`) and will do the work as soon as the other branch is merged.

Let me know (or just resolve this conversation) if that's acceptable. Or if you can think of a better way to solve it, let me know :)

Reply:
(I fixed the easy comment issue btw :sweat_smile: )

Reply:
Makes sense to me!